core/src/rpc.rs

```diff
@@ -1074,12 +1107,16 @@ impl RpcSol for RpcSolImpl {
             Some(config) if config.commitment == CommitmentLevel::Recent => 5,
             _ => 30,
         };
+        let config = commitment.map(|commitment| RpcSignatureStatusConfig {
+            commitment: Some(commitment),
+            search_transaction_history: None,
+        });
         loop {
             signature_status = meta
                 .request_processor
                 .read()
                 .unwrap()
-                .get_signature_statuses_with_commitment(vec![signature], commitment.clone())?
+                .get_signature_statuses(vec![signature], config.clone())?
```

Comment:
nit: `get_signature_statuses` could take `config` by reference to avoid cloning

Reply:
Will fix when the config direction is resolved :)
Will update docs as well.

Reply:
Fyi, clippy actually didn't like this, at least with the return to commitment:
```
error: this argument (1 byte) is passed by reference, but would be more efficient if passed by value
```

Reply:
Ah, because `config` is small now? No worries!